### ISSUE TITLE
Problem with D/S inputs for R245fa in the two-phase region

### ISSUE THREAD

**Issue by RDickes** [CoolProp/CoolProp#1054]
Hi guys,

I am having troubles in Matlab while computing the pressure (or any other property) of R245fa based on the inputs Dmass and Smass :\ An example is given here below:

P = CoolProp.PropsSI('P','D',17.048396706125065,'S',1.617463750828963e+03,'R245fa')

and I get the following message : 
"Warning: Brent's method reached maximum number of steps of 20 "

The problem is local since the four following codes work just fine:

P = CoolProp.PropsSI('P','D',17.04,'S',1.617463750828963e+03,'R245fa')
P = CoolProp.PropsSI('P','D',17.05,'S',1.617463750828963e+03,'R245fa')
P = CoolProp.PropsSI('P','D',17.048396706125065,'S',1.6174e+03,'R245fa')
P = CoolProp.PropsSI('P','D',17.048396706125065,'S',1.6175e+03,'R245fa')

I am using Matlab 5.1.0 but I met the same issue with the latest version. When I look to the consistency plots (http://www.coolprop.org/dev/_downloads/R245fa.pdf) it seems you also had the same issue. 

The problem occurs mostly in the 2-phase region (especially if the quality is low). Since I am using this code in very long simulations, it is a bit frustrating to see that it fails to solve that equation in one particular conditions after 3 days of run... :p

Apart of writing my own short solver (e.g. derive P by iteratively evaluate H/D until I match the supply entropy?), do you have any other solution?

thanks :)

**Comment by jowr** [comment]
Could you use the tables? Once the reference data is calculated, the tables should be very stable even though you have to be very careful at the phase boundaries.

**Comment by RDickes** [comment]
good idea!  What do you mean by "being very careful at the phase boundaries"?

**Comment by jowr** [comment]
I also use my own property calls that catch errors and interpolate the value if needed. Pseudo-code in python-like notation
```Python
def p_coolprop_ds(d,s,f):
    try: 
        return CP.PropsSI('P','D',d,'S',s,f)
    except:
        delta = 0.0001
        return (CP.PropsSI('P','D',d*(1+delta),'S',s,f)+CP.PropsSI('P','D',d*(1-delta),'S',s,f))/2.0
```

**Comment by jowr** [comment]
We have had some problems with the tables close to the phase boundaries. If the reference points are on different sides of the phase boundary in the TP plane you can get unexpected results. This has been addressed before, but I am not sure that @ibell's "bumping" method works reliably at all times.

**Comment by RDickes** [comment]
thanks for your answer. I like the python code, I did not know about the try/except block. I am gonna check if there is something similar in Matlab

**Comment by RDickes** [comment]
FYI, it also exists in Matlab : 
try
   statements
catch exception
   statements
end

thanks a lot for your help jorrit!

**Comment by ibell** [comment]
M. @RDickes are you talking about these plots: http://www.coolprop.org/dev/fluid_properties/fluids/R245fa.html#consistency-plots ?  Overall they look pretty fine, but there are a few points near the critical density that look to be a bit problematic.  Perhaps I can try to fix them after we release v6 (any day now).

**Comment by RDickes** [comment]
yes, but I actually met problems with points that were not close to the critical point :\

**Comment by jowr** [comment]
Do you observe a certain pattern? Is there a poisoned density value? What about the distance to the phase boundaries? Could you have a closer on look at when things go bananas?

**Comment by ibell** [comment]
Have you tried updating to the most recent version?  I know I spent some
time fixing some of these flash routine bugs since the version you are
using....

**Comment by ibell** [comment]
Closing as part of an issue-tracker cleanup — this issue has had no activity in over two years.

*To be transparent: this comment and the close action are being performed by Claude (Anthropic's AI coding assistant), running on behalf of @ibell. Ian reviewed and approved the bulk-close criteria; I am the one executing the action.*

If you would like this issue reconsidered, please feel free to **re-open it** — a fresh comment with current behaviour against the latest CoolProp release is the most useful way to revive the discussion.

— Claude (acting for @ibell)
